apps/frontend/src/app/5_pages/BitocracyPage/components/NewProposalForm/components/ProposalDataForm/ProposalDataForm.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { FC, useCallback, useMemo, useState } from 'react';
+
+import { t } from 'i18next';
+
+import {
+  Button,
+  ButtonStyle,
+  FormGroup,
+  Icon,
+  IconNames,
+  Input,
+  InputSize,
+} from '@sovryn/ui';
+
+import { translations } from '../../../../../../../locales/i18n';
+import { validateURL } from '../../../../../../../utils/helpers';
+import { FormData, ProposalType } from '../../NewProposalForm.types';
+
+export type ProposalDataFormProps = {
+  value: FormData;
+  onChange: (value: FormData) => void;
+  proposalType: ProposalType;
+  onBack: () => void;
+};
+
+export const ProposalDataForm: FC<ProposalDataFormProps> = ({
+  value,
+  onChange,
+  proposalType,
+  onBack,
+}) => {
+  const [form, setForm] = useState(value);
+
+  const isValidUrl = useMemo(
+    () => validateURL(form.discussionUrl),
+    [form.discussionUrl],
+  );
+
+  const isSubmitDisabled = useMemo(
+    () =>
+      !form.title ||
+      !form.description ||
+      !form.discussionUrl ||
+      !form.description ||
+      !isValidUrl,
+    [form.description, form.discussionUrl, form.title, isValidUrl],
+  );
+
+  const onChangeHandler = useCallback(() => onChange(form), [form, onChange]);
+
+  return (
+    <div className="flex flex-col gap-4 relative pt-4">
+      <button className="absolute left-0 -top-2" onClick={onBack}>
+        <Icon size={14} icon={IconNames.ARROW_BACK} />
+      </button>
+      <FormGroup label={t(translations.bitocracyPage.proposalDataForm.title)}>
+        <Input
+          value={form.title}
+          maxLength={140}
+          className="max-w-[22rem]"
+          onChangeText={title => setForm(form => ({ ...form, title }))}
+          size={InputSize.large}
+        />
+      </FormGroup>
+
+      <FormGroup
+        label={t(translations.bitocracyPage.proposalDataForm.discussionUrl)}
+        errorLabel={
+          isValidUrl || !form.discussionUrl
+            ? undefined
+            : t(translations.bitocracyPage.proposalDataForm.invalidURL)
+        }
+      >
+        <Input
+          value={form.discussionUrl}
+          className="max-w-[22rem]"
+          onChangeText={discussionURL =>
+            setForm(form => ({ ...form, discussionURL }))
```

Comment:
```suggestion
          onChangeText={discussionUrl =>
            setForm(form => ({ ...form, discussionUrl }))
```
Because of this, the discussion URL form field was not updated and URL validation didn't work.